### ISSUE TITLE
RqlBuilder or() not working correctly when providing multiple statements

### ISSUE THREAD

**Issue by ThomasReyskens** [ExtraHorizon/javascript-sdk#687]
## Description
RqlBuilder's `or` doesn't work when adding multiple clauses. The server replies w/ an invalid rql response.
The builder should transform the incoming string to an `and` clause to fix this

## Steps to Reproduce
1. build and rql string with multiple clauses in an or statement
2. submit request
3. error

## Code example

```typescript
const rql = rqlBuilder()
    .or(
      rqlBuilder().eq('status', 'reviewd').intermediate(),
      rqlBuilder().eq('status', 'analyzed').eq('data.indicator','normal').intermediate()
    ).build();
```

## Specifications

- Version: 7.1.1-dev.212

**Comment by SalinaExh** [comment]
https://extrahorizon.atlassian.net/browse/SVCS-449

**Comment by wwaaijer-exh** [comment]
How I believe the RQL builder is building this:
`or(status=reviewed,status=analyzed&data.indicator=normal)`
How I believe the RQL build should build this:
`or(status=reviewed,eq(status=analyzed,data.indicator=normal))`

**Comment by wwaaijer-exh** [comment]
Current workaround:
```js
rqlBuilder()
  .or(
    rqlBuilder().eq('status', 'reviewd').intermediate(),
    rqlBuilder().and(
      rqlBuilder().eq('status', 'analyzed').intermediate(),
      rqlBuilder().eq('data.indicator','normal').intermediate()
    ).intermediate()
  ).build();
```

**Comment by wwaaijer-exh** [comment]
Similar issue reported by Pieter some time ago:
```js
// Not working
  rqlBuilder().contains('patient_enlistments', rqlBuilder()
    .in('group_id', visibleGroupsIds)
    .eq('expired', 'false')
    .intermediate())
  .build();
```
Proposed workaround:
```js
rqlBuilder().contains(
  'patient_enlistments',
  rqlBuilder().in('group_id', visibleGroupsIds).intermediate(),
  rqlBuilder().eq('expired', 'false').intermediate()
).build();
```

**Comment by wwaaijer-exh** [comment]
Fixed in `8.3.0`
